test/cppsuite/src/util/barrier.cpp

```diff
@@ -39,14 +39,12 @@ void
 barrier::wait()
 {
     std::unique_lock<std::mutex> lock{_mutex};
-    auto lock_gen = _generation;
     if (!--_count) {
         _generation++;
         _count = _threshold;
         _cond.notify_all();
     } else {
-        if (!_cond.wait_for(lock, std::chrono::seconds(_sync_timeout),
-              [this, lock_gen] { return lock_gen != _generation; }))
+        if (_cond.wait_for(lock, _sync_timeout) == std::cv_status::timeout)
```

Comment:
I don't have particularly good knowledge of cppsuite, so I'm probably missing something here. This would previously return true if `_generation` changed under us, indicating that we no longer want to bother acquiring the lock, and skip the warning message. What's the motivation for removing this behaviour?

Reply:
It looks unnecessary to me: when the generation changes in the `if` statement above, all threads are unblocked via `notify_all()`. There will be a slight change of this "racy" behavior which I figure might be actually be desirable in a test suite. Also it is simpler.

Reply:
Ah, that makes sense - thanks for explaining.